packages/backend-common/src/cache/CacheManager.ts

```diff
@@ -65,14 +65,20 @@ export class CacheManager {
   }
 
   /**
-   * Generates a CacheManagerInstance for consumption by plugins.
+   * Generates a PluginCacheManager for consumption by plugins.
    *
    * @param pluginId The plugin that the cache manager should be created for. Plugin names should be unique.
    */
   forPlugin(pluginId: string): PluginCacheManager {
     return {
       getClient: (opts = {}): CacheClient => {
         const concreteClient = this.getClientWithTtl(pluginId, opts.defaultTtl);
+
+        // Attach error handler if provided.
+        if (typeof opts?.onError === 'function') {
+          concreteClient.on('error', opts.onError);
```

Comment:
I'm assuming the individual operations that are attempted will throw as well right? Meaning we will already be detecting the errors and possibly shutting down the instance if health checks fail. Tbh I'm wondering if we could just add a default implementation that just logs the errors instead? Possible even removing the option to supply your own error handler, at least for now.

Reply:
Yep that's kinda what I'm thinking too with the above comment